### DISCUSSION TITLE
Can't saveEvent With Id

### DISCUSSION THREAD

**Discussion by cormacncheese** [wmcmahan/react-native-calendar-events#372] [Q&A]
When I try to add an id to my new event I get an error: "Unknown error from a native module". I'm running it on iOS iPhone 11 Pro

I can add a basic event just fine without an id:
```
RNCalendarEvents.saveEvent(classDoc.name,
        {
          startDate: st,
          endDate: et,
        },
) 
```

but as soon as I add an id, it breaks:
```
RNCalendarEvents.saveEvent(classDoc.name,
        {
          id: 'FE6B128F-C0D8-4FB8-8FC6-D1D6BA015CDE',
          startDate: st,
          endDate: et,
        },
) 
```
Error: Unknown error from a native module

**Comment by wmcmahan** [comment]
Is the `id` that you are supplying a valid `id` of an existing event on the device? You will also need to ensure that you have permission to write to the calendar that the event with `id` belongs to.

**Comment by cormacncheese** [comment]
Figured it out. 

I misunderstood the docs and was attempting to create an event with my own custom ID.

**Comment by AndroidPayal** [comment]
having same issue. I want to give custom id to my event. So I can fetch it by Id.

**Comment by AndroidPayal** [comment]
I Understood.
For anyone stucking on same place, according to official doc saveEvent returns id of newly created event we can not externally provide id while creating new event.
We can use id field for updation of existing event.

**Comment by mfbx9da4** [comment]
Question for the maintainers. Is this a limitation of the underlying native calendar API's? Ideally we could specify our own eventId...

**Reply by kidroca** [reply]
It's limitation by the underlying native `EventKit` library: https://developer.apple.com/documentation/eventkit/ekevent/1507437-eventidentifier?language=objc. `eventIdentifier` is a readonly property

As per documentation here: https://developer.apple.com/documentation/eventkit/ekevent?language=objc
> Use the [eventWithEventStore:](https://developer.apple.com/documentation/eventkit/ekevent/1507483-eventwitheventstore?language=objc) method to create a new event. Use the properties in the class to get and modify certain information about an event

We first create an `Event` instance, then modify it's properties and finally save it, since `eventIdentifier` is read-only we can't change it
